demos/serverless/src/handlers.js

```diff
@@ -341,16 +342,98 @@ exports.end_capture = async (event, context) => {
+};
+
+
+exports.start_live_connector = async (event, context) => {
+  // Fetch the meeting by title
+  const meeting = await getMeeting(event.queryStringParameters.title);
+  meetingRegion = meeting.Meeting.MediaRegion;
+  const ivsChannelNamePrefix = IVS_CHANNEL_PREFIX?IVS_CHANNEL_PREFIX: "liveConnector"
+  const ivsChannelName = `${ivsChannelNamePrefix}-${meetingRegion}`;
+  ivsEndpoint = ''
+  rtmpURL = ''
+  ivsPlaybackUrl = ''
+  ivsChannelArn = ''
+  const params = {
+    name: ivsChannelName,
+  };
+  const ivsCreateChannelResponse = await ivs.createChannel(params).promise();
+  ivsPlaybackUrl = ivsCreateChannelResponse.channel.playbackUrl
+  ivsEndpoint = ivsCreateChannelResponse.channel.ingestEndpoint
+  ivsStreamKey = ivsCreateChannelResponse.streamKey.value
+  ivsChannelArn = ivsCreateChannelResponse.channel.arn
+  rtmpURL = "rtmps://" + ivsEndpoint + ":443/app/" + ivsStreamKey
+  console.log("Successfully created ivs channel: ", ivsCreateChannelResponse)
+  const request =
+      {
+        Sinks: [
+          {
+            RTMPConfiguration: {
+              AudioChannels: "Stereo",
+              AudioSampleRate: "48000",
+              Url: rtmpURL
+            },
+            SinkType: "RTMP"
+          }
+        ],
+        Sources: [
+          {
+            ChimeSdkMeetingLiveConnectorConfiguration: {
+              Arn: `arn:aws:chime::${AWS_ACCOUNT_ID}:meeting:${meeting.Meeting.MeetingId}`,
+              CompositedVideo: {
+                GridViewConfiguration: {
+                  ContentShareLayout: "Vertical",
+                },
+                Layout: "GridView",
+                Resolution: "FHD",
+              },
+              MuxType: "AudioWithCompositedVideo"
+            },
+            SourceType: "ChimeSdkMeeting"
+          }
+        ]
+      };
+  console.log("Creating new media live connector pipeline: ", request)
+  liveConnectorPipelineInfo = await chimeSdkMediaPipelines.createMediaLiveConnectorPipeline(request).promise();
+  await putLiveConnectorPipeline(event.queryStringParameters.title, liveConnectorPipelineInfo)
+  await putIvsArn(event.queryStringParameters.title, ivsChannelArn)
+  console.log("Successfully created media live connector pipeline: ", liveConnectorPipelineInfo)
+  liveConnectorPipelineInfo["playBackUrl"] = ivsPlaybackUrl
+  return response(201, 'application/json', JSON.stringify(liveConnectorPipelineInfo));
+};
+
+exports.end_live_connector = async (event, context) => {
+  // Fetch the live connector pipeline info by title
+  const liveConnectorPipelineInfo = await getLiveConnectorPipeline(event.queryStringParameters.title);
+  if (liveConnectorPipelineInfo) {
+    ivsCreated = false
+    const ivsChannelArn = await getIvsArn(event.queryStringParameters.title)
+    await getClientForMediaCapturePipelines().deleteMediaPipeline({
+      MediaPipelineId: liveConnectorPipelineInfo.MediaLiveConnectorPipeline.MediaPipelineId
+    }).promise();
+
+    try {
+      await delay(3000);
```

Comment:
Any reason for this delay here?

Reply:
because when delete the live connector pipeline, there is some delay to stop the stream. If stream hasn't been stopped, then delete ivs channel will fail. 3 second is more than enough.